itests/hive-minikdc/src/test/java/org/apache/hive/minikdc/MiniHiveKdc.java

```diff
@@ -194,9 +194,6 @@ public static MiniHS2 getMiniHS2WithKerb(MiniHiveKdc miniHiveKdc, HiveConf hiveC
                                               .withConf(hiveConf)
                                               .withMiniKdc(hivePrincipal, hiveKeytab)
                                               .withAuthenticationType(authType);
-    if (HiveServer2.isHttpTransportMode(hiveConf)) {
```

Comment:
why do we remove it here? what if the HS2 reads this property and opens the http server, while the client is trying to establish the connection with the binary port?

Reply:
The JDBC URL is usually obtained from `MiniHS2#getJdbcURL()` so the server/client configs should be aligned. 

Nevertheless to simplify the review of this PR I reverted the changes here and in MiniHS2 and added the same `isHttpTransportMode` logic in the the other static factory method that is used in this test (https://github.com/apache/hive/pull/5315/commits/add40a8b80d2be8ce5f03022c4206259bdb421de ).